mmdet3d/datasets/s3dis_dataset.py

```diff
@@ -0,0 +1,326 @@
+import numpy as np
+from os import path as osp
+
+from mmdet3d.core import show_seg_result
+from mmdet.datasets import DATASETS
+from .custom_3d_seg import Custom3DSegDataset
+from .pipelines import Compose
+
+
+@DATASETS.register_module()
+class _S3DISSegDataset(Custom3DSegDataset):
+    """S3DIS Dataset for Semantic Segmentation Task.
+
+    This class is the inner dataset for S3DIS. Since S3DIS has 6 areas, we
+    often train on 5 of them and test on the remaining one. There should be an
+    wrapper concating all the provided data in different areas.
```

Comment:
How much difference are the dataset classes created for these areas? What makes the inner class necessary?

Reply:
Not much difference as the wrapper class inherits the inner class. My intuition is that one dataset instance should only process one data_info file. I agree it's not necessary and I can write them into one class. I just think it's more clear to have one dataset for one area and use a wrapper to concat them?

Reply:
Maybe you can just read infos just in the init? From my perspective, both are ok indeed except that there exists some inconsistency with other dataset class. How do you think about it @ZwwWayne ?

Reply:
After discussing with Wenwei, he thinks we can leave it as it. And I have added some comments to explain it in the code.